src/OrleansRuntime/Catalog/DefaultGrainActivator.cs

```diff
@@ -0,0 +1,62 @@
+﻿using System;
+using System.Collections.Concurrent;
+using System.Globalization;
+using Microsoft.Extensions.DependencyInjection;
+
+namespace Orleans.Runtime
+{
+    /// <summary>
+    /// <see cref="IGrainActivator"/> that uses type activation to create grains.
+    /// </summary>
+    public class DefaultGrainActivator : IGrainActivator
```

Comment:
I was about to suggest that maybe `IGrainFactory` was a more suitable name. Nevermind... 😕 would `IGrainCreator` make sense, since this doesn't actually call the activation lifecycle method on the grain?